program-manager/src/domain/cosmos_cw.rs

```diff
@@ -133,6 +134,8 @@ impl CosmosCosmwasmConnector {
             code_ids: code_ids.clone(),
             chain_name: chain_info.name.clone(),
             prefix: chain_info.prefix.clone(),
+            api: cosmwasm_std::testing::MockApi::default()
+                .with_prefix(Box::leak(Box::new(chain_info.prefix.clone()))),
```

Comment:
I can't think of a way where this can go wrong but it'd be nice to add some comments about why `Box::leak` is fine here. Guess we could also have a chain prefix enum (with to/from string impl) which would make it static so we wouldn't need to deal with boxing like this?

Reply:
Thanks @bekauz and @stiiifff 
We have a weird play with the connectors, because they are not cached for the entire program, because the manager is not a program, rather a library, so they are cached for the functions (init , update, migrate).
This was leaking the prefix on every time we call a function, not much, and can easily fixed with a server restart, but still, it is a leak.

I just copied the MockApi from cosmwas_std, and changed the api to accept String instead of the static str, which doesn't leak memory now.

Reply:
also allowed us to remove our helpers for it and just use the mock api.